### PR TITLE
Prohibit closing the write index for a data stream

### PR DESCRIPTION
**Pull request by danhermann**
Relates to #53100

### CODE REVIEW

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexStateService.java [hunk 1]
```diff
@@ -137,6 +138,19 @@ public void closeIndices(final CloseIndexClusterStateUpdateRequest request, fina
         if (concreteIndices == null || concreteIndices.length == 0) {
             throw new IllegalArgumentException("Index name is required");
         }
+        List<String> writeIndices = new ArrayList<>();
+        SortedMap<String, IndexAbstraction> lookup = clusterService.state().metadata().getIndicesLookup();
+        for (Index index : concreteIndices) {
+            IndexAbstraction ia = lookup.get(index.getName());
+            if (ia != null && ia.getParentDataStream() != null &&
+                ia.getParentDataStream().getWriteIndex().getIndex().getName().equals(index.getName())) {
```
Comment (martijnvg) [inline server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexStateService.java:146]: I think we should change this to:

```
ia.getParentDataStream().getWriteIndex().getIndex().equals(index))
```

The index uuid should be the same.

**Review by martijnvg — APPROVED** [review]
Left one comment, otherwise LGTM

**Pull request merged**